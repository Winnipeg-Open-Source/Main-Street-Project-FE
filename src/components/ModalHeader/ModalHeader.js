import React from 'react';
import PropTypes from 'prop-types';
import { CloseButton, Flex, Text } from 'pcln-design-system';

function ModalHeader ({ className, title, subtitle, onClose }) {
    return (
        <Flex
            className={className}
            flexDirection='column'
            color='primary.base'
            height='auto'
            p={3}
        >
            <Flex justifyContent='space-between'>
                <Text bold>{title}</Text>
                {onClose && (
                    <CloseButton onClick={onClose} data-testid='close-icon' />
                )}
            </Flex>
            <Text fontSize={1}>{subtitle}</Text>
        </Flex>
    );
}

ModalHeader.displayName = 'ModalHeader';

ModalHeader.propTypes = {
    className: PropTypes.string,
    title: PropTypes.string,
    subtitle: PropTypes.string,
    onClose: PropTypes.func,
};

ModalHeader.defaultProps = {
    className: '',
};

export default ModalHeader;
