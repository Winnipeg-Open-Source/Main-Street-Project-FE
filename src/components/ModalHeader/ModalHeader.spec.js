import React from 'react';
import { fireEvent } from '@testing-library/react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import ModalHeader from './ModalHeader';

describe('ModalHeader', () => {
    it('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<ModalHeader />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders with a title', () => {
        const { queryByText, queryByTestId } = renderWithTheme(
            <ModalHeader title='Title' />
        );

        expect(queryByText(/Title/)).toBeInTheDocument();
        expect(queryByText(/Subtitle/)).not.toBeInTheDocument();
        expect(queryByTestId('close-icon')).not.toBeInTheDocument();
    });

    it('renders with a subtitle', () => {
        const { queryByText, queryByTestId } = renderWithTheme(
            <ModalHeader title='Title' subtitle='Subtitle' />
        );

        expect(queryByText(/Title/)).toBeInTheDocument();
        expect(queryByText(/Subtitle/)).toBeInTheDocument();
        expect(queryByTestId('close-icon')).not.toBeInTheDocument();
    });

    it('renders a close button', () => {
        const mockOnClose = jest.fn();
        const { queryByText, queryByTestId } = renderWithTheme(
            <ModalHeader
                title='Title'
                subtitle='Subtitle'
                onClose={mockOnClose}
            />
        );

        const closeButton = queryByTestId('close-icon');

        expect(queryByText(/Title/)).toBeInTheDocument();
        expect(queryByText(/Subtitle/)).toBeInTheDocument();
        expect(closeButton).toBeInTheDocument();

        expect(mockOnClose).toHaveBeenCalledTimes(0);
        fireEvent.click(closeButton);
        expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
});
