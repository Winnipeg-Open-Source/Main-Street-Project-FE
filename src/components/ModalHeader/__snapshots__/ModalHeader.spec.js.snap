// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-AxhCb fidhWr"
      color="primary.base"
      height="auto"
    >
      <div
        class="sc-AxirZ sc-AxhCb dkAhFW"
      >
        <div
          class="sc-AxhUy gGlGcm"
        />
      </div>
      <div
        class="sc-AxhUy hPygOl"
        font-size="1"
      />
    </div>
  </div>
</DocumentFragment>
`;
